adaptdl/adaptdl/torch/_metrics.py

```diff
@@ -105,24 +103,27 @@ def get_goodput_fn():
 
 def _fit_perf_params():
     state = _metrics_state()
-    items = state.profile.items()
-    items = [item for item in items if item[1]["count"] > 0]
-    keys = [item[0] for item in items]
-    values = [item[1] for item in items]
-    num_nodes, num_replicas, local_bsz, accumulation_steps = \
-        (np.array(val) for val in zip(*keys))
-    step_time = np.array([val["step_time"] / val["count"] for val in values])
-    sync_time = np.array([val["sync_time"] / val["count"] for val in values])
-    accumulation_time = np.array(
-            [val["accumulation_step_time"] / val["accumulation_count"]
-             if val["accumulation_count"] > 0 else 0.0
-             for val in values])
-    compute_time = step_time - sync_time
-    accumulation_time = np.where(
-        accumulation_steps > 0, accumulation_time, compute_time)
-    state.perf_params = fit_perf_params(
-        num_nodes, num_replicas, local_bsz, accumulation_steps,
-        step_time, compute_time, accumulation_time)
+    profile = {k: v for k, v in state.profile.items() if v.get("optim_count")}
+    # Convert profile into numpy arrays.
+    num_nodes, num_replicas, atomic_bsz = (np.array(k) for k in zip(*profile))
+    accum_step_time = np.array([v.get("accum_step_time", 0.0)
+                                for v in profile.values()])
+    accum_count = np.array([v.get("accum_count", 0) for v in profile.values()])
+    optim_step_time = np.array([v.get("optim_step_time", 0.0)
+                                for v in profile.values()])
+    optim_sync_time = np.array([v.get("optim_sync_time", 0.0)
+                                for v in profile.values()])
+    optim_count = np.array([v.get("optim_count", 0) for v in profile.values()])
+    assert np.all(optim_count > 0)
```

Comment:
Is a hard error the right call here? This will kill the adaptdljob entirely right?

Reply:
It should never happen because of line 106. Add the assert here to catch future programming errors